### PR TITLE
[1.1.x] SLIM_LCD_MENUS option

### PR DESCRIPTION
**Pull request by houseofbugs**
Saving Space for 1284P (and other small) CPUs. Saves about 8K of memory. More to come.

[Concise Diff](8947/files?w=1)

### CODE REVIEW

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -3456,12 +3462,16 @@ void kill_screen(const char* lcd_msg) {
     //
     // Preheat Material 1 conf
     //
-    MENU_ITEM(submenu, MSG_PREHEAT_1_SETTINGS, lcd_control_temperature_preheat_material1_settings_menu);
+	#if DISABLED(SLIM_1284P)    
+		MENU_ITEM(submenu, MSG_PREHEAT_1_SETTINGS, lcd_control_temperature_preheat_material1_settings_menu);
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp]: No tabs, buddy.

File: Marlin/Configuration.h [hunk 2]
```diff
@@ -383,7 +383,9 @@
 // If your configuration is significantly different than this and you don't understand the issues involved, you probably
 // shouldn't use bed PID until someone else verifies your hardware works.
 // If this is enabled, find your own PID constants below.
-//#define PIDTEMPBED
+#if DISABLED(SLIM_1284P)
+  #define PIDTEMPBED
+#endif
```
Comment (thinkyhead) [inline Marlin/Configuration.h]: That won't work.

File: Marlin/Configuration_adv.h [hunk 3]
```diff
@@ -34,6 +34,19 @@
...
+//===========================================================================
+//=============================    Slimming      ============================
+//===========================================================================
+
+// Turns off un-needed features to save space on limited boards like the Melzi/Creality/Anet type boards with the 1284P CPU
+//#define SLIM_1284P
+
+#if ENABLED(SLIM_1284P)
+  #define NO_VOLUMETRICS
+#endif
```
Comment (thinkyhead) [inline Marlin/Configuration_adv.h]: This won't work either. Things like this must be done in `Conditionals_post.h`.

**Comment by fiveangle** [comment]
@houseofbugs - thanks for putting the effort into this as we are at a point where there is no option for some of these smaller boards but to start reducing functionality.  But doesn't it make sense to add some detail in the comment for the feature what functionality is lost ?  The alternative is user enables the feature and then pokes around to see what's changed.  If something is missing they don't know if it should be missing or not, so is indistinguishable from a bug.

-=dave

**Comment by fiveangle** [comment]
Is this supposed to disable X/Y/Z moment menus entirely ?  Those are fundemental to having an LCD.  They are gone with this feature enabled.

**Comment by thinkyhead** [comment]
> Is this supposed to disable X/Y/Z moment menus entirely?

After you home with `G28` the submenu will appear (assuming `DELTA`).

**Comment by thinkyhead** [comment]
> doesn't it make sense to add some detail in the comment for the feature what functionality is lost?

Couldn't hurt.

**Pull request merged**